app/components/SearchableMetricSelector.js

```diff
@@ -0,0 +1,61 @@
+import { useEffect, useState } from 'react'
+
+function Modal(props){
+  const { supportedMetrics, onClose, selectMetric, show } = props;
+  if(!show){
+    return null;
+  }
+  
+  return (
+    <div className="SelectorModal" onClick={onClose} >
+        <div className="modal-content" onClick={e => e.stopPropagation() } >
```

Comment:
Great job figuring out how to make the modal open and close on your own, especially the e.stopPropagation() part to make a click on a child element not fire the click handler in the parent element is a tricky thing to learn the first time.